tests/simplex/refinement_05.cc

```diff
@@ -0,0 +1,62 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+
+
+// Test refinement of 3D simplex mesh.
+#include <deal.II/base/quadrature_lib.h>
+
+#include <deal.II/fe/fe_simplex_p.h>
+#include <deal.II/fe/mapping_fe.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+
+#include "../tests.h"
+
+
+
+void
+test()
+{
+  constexpr int dim = 3;
+
+  for (unsigned int i = 0; i < 3; i++)
+    {
+      Triangulation<dim> tria;
+      GridGenerator::subdivided_hyper_cube_with_simplices(tria, 1, 0, 1);
+      if (i > 0)
+        tria.refine_global(i);
```

Comment:
I think technically you can simply write
```suggestion
      tria.refine_global(i);
```
as passing in zero refinements will simply return immediately and not do anything. As a second comment, you could also create `tria` outside the `for` loop over `i` and then call `refine_global` once per loop iteration (you could do it in the end of the loop if you want to start with index 0). None of this is important, as it is only a test file.

Reply:
Changed the ordering to minimize the number of times the refinement has to be done.